File: app/src/main/java/foo/bar/musicplayer/SearchListAdapter.java
package foo.bar.musicplayer;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import foo.bar.musicplayer.model.Artist;
import foo.bar.musicplayer.model.Artists;

public class SearchListAdapter extends RecyclerView.Adapter<SearchListAdapter.ListingViewHolder> {

    private static final String TAG = SearchListAdapter.class.getSimpleName();

    private Artists artists = new Artists();

    private Context context;

    public SearchListAdapter(Context context) {
        this.context = context;
    }

    public void setSearchResults(Artists artists) {
        this.artists = artists;
        notifyDataSetChanged();
    }

    @Override
    public SearchListAdapter.ListingViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new ListingViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.artist_search_result_item, parent, false));
    }

    @Override
    public void onBindViewHolder(SearchListAdapter.ListingViewHolder holder, int position) {
        holder.bind(artists.getItems().get(position));
    }

    @Override
    public int getItemCount() {
        if (artists != null) {
            return artists.getItems().size();
        } else {
            return 0;
        }
    }

    static class ListingViewHolder extends RecyclerView.ViewHolder {

        private TextView artistName;
        public ListingViewHolder(View itemView) {
            super(itemView);
            artistName = (TextView) itemView.findViewById(R.id.artist_name);
        }

        public void bind(Artist artist){
            artistName.setText(artist.getName());
        }
    }
}

File: app/src/main/java/foo/bar/musicplayer/ui/search/SearchListAdapter.kt
package foo.bar.musicplayer.ui.search

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import foo.bar.musicplayer.R
import foo.bar.musicplayer.model.Artist
import foo.bar.musicplayer.util.ImageUtils
import kotlinx.android.synthetic.main.artist_search_result_item.view.*

class SearchListAdapter : RecyclerView.Adapter<SearchListAdapter.ListingViewHolder>() {

  private var artists = mutableListOf<Artist>()

  fun setSearchResults(artists: List<Artist>) {
    this.artists.clear()
    this.artists.addAll(artists)
    notifyDataSetChanged()
  }

  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): SearchListAdapter.ListingViewHolder {
    val layoutInflater = LayoutInflater.from(parent.context)
    val view = layoutInflater.inflate(R.layout.artist_search_result_item, parent, false)
    return ListingViewHolder(view)
  }

  override fun onBindViewHolder(holder: SearchListAdapter.ListingViewHolder, position: Int) {
    holder.bind(artists[position])
  }

  override fun getItemCount() = artists.size

  class ListingViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    fun bind(artist: Artist): Unit = with(itemView) {
      ImageUtils.setImage(context, r_result_img, artist.images)

      r_result_name.text = artist.name
      r_result_popularity.text = artist.popularity
    }
  }

  companion object {
    private val TAG = SearchListAdapter::class.java.simpleName
  }
}
